lib/karafka/instrumentation/vendors/datadog/logger_listener.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+module Karafka
+  module Instrumentation
+    # Namespace for vendor specific instrumentation
+    module Vendors
+      # Datadog specific instrumentation
+      module Datadog
+        # A karafka's logger listener for Datadog
+        # It depends on the 'ddtrace' gem
+        class LoggerListener
+          include ::Karafka::Core::Configurable
+          extend Forwardable
+
+          def_delegators :config, :client
+
+          # `Datadog::Tracing` client that we should use to trace stuff
+          setting :client
+
+          configure
+
+          # Log levels that we use in this particular listener
+          USED_LOG_LEVELS = %i[
+            info
+            error
+            fatal
+          ].freeze
+
+          private_constant :USED_LOG_LEVELS
+
+          # @param block [Proc] configuration block
+          def initialize(&block)
+            configure
+            setup(&block) if block
```

Comment:
Hey @mensfeld 👋 

Can you please explain me why do we need both the `configure` and the `setup(&block)` that calls `configure(&block)`?

My guess is that you're covering two use cases:
1. when someone instantiates this class and only afterwards changes the `client` or `config`.
2. when someone instantiates this class and passes a block to set up `client` or `config`.

Is that right?

Reply:
Yes exactly. This is to align the API with other components.